### DISCUSSION TITLE
Flexible configuration without prefix for English language

### DISCUSSION THREAD

**Discussion by ansbr** [ivanhofer/typesafe-i18n#646] [Q&A] [answered]
Hi! I have a website that I want to rewrite using Svelte Kit, but in your example with typesafe-i18n for Svelte Kit, it's not possible to remove the prefix for the English language. I don't want to rewrite the current link structure and keep the same URL system.
English: https://msng.link/generator/
Russian: https://msng.link/ru/generator/
Spanish: https://msng.link/es/generator/
German: https://msng.link/de/generator/

Is it possible to do this? If yes, what changes need to be made to your example for Svelte Kit?

While I was writing the message, I saw that the user @pavelstianko made such an implementation https://github.com/ivanhofer/typesafe-i18n/discussions/630. But there is no example, I hope he will answer and help me do the same, or someone else will suggest a solution. Thank you!

**Comment by ivanhofer** [comment, marked as answer]
@ansbr you need to use [optional route parameter](https://kit.svelte.dev/docs/advanced-routing#optional-parameters).
And then you need to remove the redirect in the `hooks.config.js`. You probably also need to make some adjustments to the `LocalSwitcher` component

**Reply by ansbr** [reply]
Thank you for answer, I think I need to learn Svelte Kit docs one more time...
